src/pydicom/dataset.py

```diff
@@ -2795,6 +2795,8 @@ def set_pixel_data(
         arr: "numpy.ndarray",
         photometric_interpretation: str,
         bits_stored: int,
+        *,
+        new_instance_uid: bool = True,
```

Comment:
The argument name sounds like a UID to me rather than a bool, i.e. the user is passing the new UID they have already generated.  Perhaps `generate_instance_uid` or something like that?

Reply:
Done